pkg/k8s/factory_functions.go

```diff
@@ -133,6 +144,15 @@ func EqualV1Endpoints(ep1, ep2 *types.Endpoints) bool {
 		reflect.DeepEqual(ep1.Subsets, ep2.Subsets)
 }
 
+func EqualV1EndpointSlice(ep1, ep2 *types.EndpointSlice) bool {
+	// We only care about the Name, Namespace and Subsets of a particular
+	// endpoint.
+	return ep1.Name == ep2.Name &&
```

Comment:
Please add a comment that `AddressType` is omitted because it immutable and currently ignored.